src/store/products/products.types.ts

```diff
@@ -0,0 +1,45 @@
+// Entities
+
+export type ProductId = number;
+
+export interface Product {
+  product_id: ProductId;
+  name: string;
+  purchase_cost: number;
+}
+
+// Action types
+
+export const PRODUCTS_FETCH_SUCCESS = "@@products/FETCH_SUCCESS";
+export const PRODUCTS_FETCH_BY_ID_SUCCESS = "@@products/FETCH_BY_ID_SUCCESS";
+export const PRODUCTS_ADD_SUCCESS = "@@products/ADD_SUCCESS";
```

Comment:
Here the '@@' I've never seen. Intuitively I would have imagined we would need to use the environment variable for the API here, that we used in defining the hook 'useAddProduct'. OR does the @@ act as a sort of placeholder for that.

In any case does the upper case (:P pun) sort of echo SQL and signify that these action types are at the fundamental level of getting data from the back-end's endpoints in pure form, whereas other calls to the data in the front-end will be from layers built on top of this one?